lib/error.js

```diff
@@ -2,9 +2,47 @@ var makeError = require('make-error');
 
 function ShareDBError(code, message) {
   ShareDBError.super.call(this, message);
-  this.code = code;
+  this.code = code || ShareDBError.code.UNKNOWN_ERROR;
```

Comment:
Let's make `this.code` all caps `this.CODES`, because it's a constant, and it helps consumers to understand it's an object.